gateway/starknet.go

```diff
@@ -30,6 +30,9 @@ func (sg *Gateway) Call(ctx context.Context, call types.FunctionCall, blockHashO
 	if len(call.Calldata) == 0 {
 		call.Calldata = []string{}
 	}
+	if len(call.Signature) == 0 {
+		call.Signature = []string{}
+	}
```

Comment:
I do not think it is useful. See https://go.dev/play/p/dM5IPzV97os for a playground. If Calldata is a `[]string` the default value is []string{}, no?

Reply:
I believe the issue is a nil string will be omitted from the encoded json and gateway wants an empty string

Reply:
oh! I'll check that too.

Reply:
You are right @tarrencev ! Sorry for the comment but I learned something.

Reply:
Correct, it’s pretty ugly but the api rejects the nil string signature slice. Open to a better default.